libraries/AP_UAVCAN/AP_UAVCAN.cpp

```diff
@@ -180,14 +179,27 @@ void AP_UAVCAN::init(uint8_t driver_index)
 
     hw_version.major = AP_UAVCAN_HW_VERS_MAJOR;
     hw_version.minor = AP_UAVCAN_HW_VERS_MINOR;
-    _node->setHardwareVersion(hw_version);
 
-    int start_res = _node->start();
-    if (start_res < 0) {
-        debug_uavcan(1, "UAVCAN: node start problem, error %d\n\r", start_res);
-        return;
-    }
+    const uint8_t uid_buf_len = hw_version.unique_id.capacity();
+    uint8_t uid_len = uid_buf_len;
+    uint8_t unique_id[uid_buf_len];
+
+    if (hal.util->get_system_id_unformatted(unique_id, uid_len)) {
+        uavcan::copy(unique_id, unique_id + uid_len, hw_version.unique_id.begin());
+
```

Comment:
This `if` needs to be closed here, otherwise systems without system_id (like Linux) won't have UAVCAN.

Reply:
I've added an implementation for get_system_id for HAL_Linux now

Reply:
I still find this to be a bug. Having a system ID is not mandatory to have UAVCAN or even the dynamic ID server.

Reply:
arguably, yes, although with this patch all HALs that support CAN do have an implementation. So it's not a serious issue :)